corehq/motech/repeaters/tasks.py

```diff
@@ -30,34 +36,27 @@
 )
 def check_repeaters():
     start = datetime.utcnow()
-    cutoff = start + CHECK_REPEATERS_INTERVAL
+    repeater_lock_timeout = RECORD_LOCK_TIMEOUT // 2
+    timeout_dt = start + timedelta(seconds=repeater_lock_timeout)
 
-    # Timeout for slightly less than periodic check
+    # Long timeout to allow all waiting repeat records to be iterated
     check_repeater_lock = get_redis_lock(
         CHECK_REPEATERS_KEY,
-        timeout=CHECK_REPEATERS_INTERVAL.seconds - 10,
+        timeout=repeater_lock_timeout,
         name=CHECK_REPEATERS_KEY,
     )
     if not check_repeater_lock.acquire(blocking=False):
         return
 
     for record in iterate_repeat_records(start):
-        now = datetime.utcnow()
-        lock_key = _get_repeat_record_lock_key(record)
-
-        if now > cutoff:
+        if not _soft_assert(
```

Comment:
nit: the negation makes it hard to understand the condition (I didn't even know `SoftAssert.__call__` returned the assertion condition). What do you think of restructuring this as:
```py
if datetime.utcnow() > timeout_dt:
    _soft_assert(False, "...")
    break
```

Reply:
OK. Fair enough. I kinda love that soft_assert [can be used in conditionals](https://github.com/dimagi/commcare-hq/blob/master/corehq/util/tests/test_soft_assert.py#L35) but grepping the codebase makes me think I'm alone in that.